### PR TITLE
Improve and fix searching behaviors

### PR DESCRIPTION
**Pull request by denkeni**
1. Tapping board search result will show board page, based users' feedback.
2. Fix search results may be incorrect when changing keywords (race condition), as attached.

https://github.com/Ptt-official-app/Ptt-iOS/assets/2727287/3dd8e94e-5803-4975-8178-3c4964c5697c

### CODE REVIEW

File: Ptt/Common/BoardSearch/BoardSearchViewController.swift [hunk 1]
```diff
@@ -72,16 +72,21 @@ final class BoardSearchViewController: UITableViewController {
 
     override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
         tableView.deselectRow(at: indexPath, animated: true)
-#if DEVELOP // Disable editing, for now
+        // Based on "Ptt 官方 app 測試小組" unofficial survey in 2023-08-14,
+        // voting for expected behavior is
+        // 8:0 for showBoardView : addBoardToFavorite
```
Comment (xxi511) [inline Ptt/Common/BoardSearch/BoardSearchViewController.swift:77]: IMO, open a board is the expected behavior..... I probably had brain fog at that time :(   
anyway ... do we need these comments?

File: Ptt/Common/BoardSearch/BoardSearchViewController.swift [hunk 2]
```diff
@@ -72,16 +72,21 @@ final class BoardSearchViewController: UITableViewController {
...
+        let boardInfo = boards[indexPath.row]
+        delegate?.boardDidSelect(info: boardInfo)
+
+        // Reserved editing feature when tapping favoriteButton
```
Comment (xxi511) [inline Ptt/Common/BoardSearch/BoardSearchViewController.swift:81]: Favorite button will be added in another PR?

File: Ptt/Common/BoardSearch/BoardSearchViewController.swift [hunk 3]
```diff
@@ -121,7 +126,7 @@ extension BoardSearchViewController {
                     if let index = response.next_idx {
                         self?.startIdx = index
                     }
-                    self?.boards += response.list
+                    self?.boards = response.list
```
Comment (xxi511) [inline Ptt/Common/BoardSearch/BoardSearchViewController.swift:129]: Correct if I was wrong

Returned boards has a maximum number in each API response   
`+=` is for pagination purpose    
`=` seems like doesn't work if the search result more than the maximum number 

But since the default number is `200`, this is an edge case

**Comment by denkeni** [comment]
@xxi511 The code here is outdated, and I would like to close it soon.

I've created https://github.com/Ptt-official-app/Ptt-iOS/issues/83 for tracking this issue. Would you like to be assigned for that? ;)

**Comment by xxi511** [comment]
what do you mean outdated ? conflicts ?

**Comment by denkeni** [comment]
> what do you mean outdated ? conflicts ?

Yep, in https://github.com/Ptt-official-app/Ptt-iOS/pull/77 `BoardSearchDelegate` was replaced by `FavoriteBoardManagerProtocol` probably, so this PR requires a total rewrite anyway.

**Comment by xxi511** [comment]
There you go https://github.com/Ptt-official-app/Ptt-iOS/pull/84
